imports/components/Tour.js

```diff
@@ -149,32 +130,72 @@ class Tour extends React.Component {
   }
 }
 
-// load 1 tour & 1+ steps
-const loadMeteorData = Component => createContainer(
-  ({ params: { user, repoName } }) => {
-    const repository = `${user}/${repoName}`;
-    const handleTour = Meteor.subscribe('tours');
-    const handleSteps = Meteor.subscribe('steps', repository);
-    return {
-      tour: Tours.findOne({ repository: repository }),
-      tourLoaded: handleTour.ready(),
-      steps: Steps.find({ tourName: repository }).fetch(),
-      stepsLoaded: handleSteps.ready(),
-    };
-  },
-  Component
+const withTour = graphql(
+  gql`
+  query getTourWithSteps($tourRepository: String!) {
+    tour(tourRepository: $tourRepository) {
+      _id
+      targetRepository
+      description
+      repository
+      steps {
+        _id
+        title
+        slug
+      }
+    }
+  }
+`,
+  {
+    options: ({ params: { user, repoName } }) => ({
+      variables: { tourRepository: `${user}/${repoName}` },
+    }),
+    props: ({ data: { loading, tour }, ownProps: { location } }) => ({ loading, tour, location }),
+  }
 );
 
 // show loading component if the tour & steps data are loading
 const displayLoadingState = branch(
-  props => !props.tourLoaded || !props.stepsLoaded,
+  props => props.loading,
   renderComponent(withProps(() => ({ big: true, statusId: 'loading' }))(ParrotSays))
 );
 
 // show not found component if no tour found or no steps found for this tour
 const displayNotFoundState = branch(
-  props => !props.tour || !props.steps.length,
+  props => !props.tour || !props.tour.steps.length,
   renderComponent(withProps(() => ({ big: true, statusId: 'not-found' }))(ParrotSays))
 );
 
-export default compose(loadMeteorData, displayLoadingState, displayNotFoundState, pure)(Tour);
+// add mutation hoc after the other hoc: the tour is actually there
+const withMutation = graphql(
+  gql`
+  mutation importTour($tourRepository: String!) {
+    importTour(tourRepository: $tourRepository) {
+      _id
+      targetRepository
+      description
+      repository
+      steps {
+        _id
+        title
+        slug
+      }
+    }
+  }
+`,
+  {
+    props: ({ ownProps: { tour: { repository } }, mutate }) => ({
+      importTour: () => mutate({
+        variables: { tourRepository: repository },
+        // basic *forced* update, we shouldn't do that normally (automatic update)
+        // note: tried without, it's not really updating the ui, maybe it's because
+        // of how we are doing our mutation?
+        refetchQueries: ['getTourWithSteps'],
```

Comment:
💡 makes sense! we remove and then insert the tour, so the `_id` changes and so the store freaks out! 🌮 🎉  fixing that.